v2/internal/controllers/crd_compute_image_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+
+	compute2020 "github.com/Azure/azure-service-operator/v2/api/compute/v1alpha1api20200930"
+	compute2021 "github.com/Azure/azure-service-operator/v2/api/compute/v1alpha1api20210701"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_Compute_Image_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	tc.LogSection("Create Resource Group")
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	tc.LogSection("Create Snapshot")
+	snapshot := &compute2020.Snapshot{
+		ObjectMeta: tc.MakeObjectMeta("snapshot"),
+		Spec: compute2020.Snapshots_Spec{
+			CreationData: compute2020.CreationData{
+				CreateOption: compute2020.CreationDataCreateOptionEmpty,
+			},
+			DiskSizeGB: to.IntPtr(32),
+			Location:   tc.AzureRegion,
+			Owner:      testcommon.AsOwner(rg),
+		},
+	}
+
+	tc.CreateResourceAndWait(snapshot)
+	tc.Expect(snapshot.Status.Id).ToNot(BeNil())
+	snapshotARMId := *snapshot.Status.Id
+
+	tc.LogSection("Create Image")
+	v2 := compute2021.ImagePropertiesHyperVGenerationV2
+	image := &compute2021.Image{
+		ObjectMeta: tc.MakeObjectMeta("image"),
+		Spec: compute2021.Images_Spec{
+			HyperVGeneration: &v2,
+			Location:         tc.AzureRegion,
+			Owner:            testcommon.AsOwner(rg),
+			StorageProfile: &compute2021.ImageStorageProfile{
+				OsDisk: &compute2021.ImageOSDisk{
+					DiskSizeGB: to.IntPtr(32),
+					OsType:     compute2021.ImageOSDiskOsTypeLinux,
+					OsState:    compute2021.ImageOSDiskOsStateGeneralized,
+					Snapshot: &compute2021.SubResource{
+						Reference: &genruntime.ResourceReference{
+							ARMID: snapshotARMId,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tc.CreateResourceAndWait(image)
+	tc.Expect(image.Status.Id).ToNot(BeNil())
+	imageARMId := *image.Status.Id
+
```

Comment:
minor: Most tests we try to do a  patch/update section too (just to ensure patch/update works and catch any possible issues in the RP like what @babbageclunk found where they didn't accept PUT for update). See for example the `crd_networking_publicip_test.go`

Reply:
I didn't find any properties that could be patched after the initial create.

Reply:
You can almost always patch `Tags`, but it's not all that interesting since we use that trick in lots of other places already so it's unlikely it's going to be broken for just this resource.